### PR TITLE
Fix #1712 and raise a warning when using map of ufuncs

### LINKED ISSUE #1712: Strange behaviour of `map` with `np.log`
Simple example:
>sss = hs.signals.Signal2D(np.arange(9).reshape(3,3))
sss.map(np.log)

>---------------------------------------------------------------------------
>ValueError                                Traceback (most recent call last)
<ipython-input-12-00b966ccb30a> in <module>()
----> 1 sss.map(np.log)

>/usr/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/hyperspy-1.3-py3.6-macosx-10.10-x86_64.egg/hyperspy/signal.py in map(self, function, show_progressbar, parallel, inplace, ragged, **kwargs)
   3506         # iterate over the coordinates.
   3507         try:
-> 3508             fargs = inspect.signature(function).parameters.keys()
   3509         except TypeError:
   3510             # This is probably a Cython function that is not supported by

>/usr/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py in signature(obj, follow_wrapped)
   2999 def signature(obj, *, follow_wrapped=True):
   3000     """Get a signature object for the passed callable."""
-> 3001     return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
   3002 
   3003 

>/usr/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py in from_callable(cls, obj, follow_wrapped)
   2749         """Constructs Signature for the given callable object."""
   2750         return _signature_from_callable(obj, sigcls=cls,
-> 2751                                         follow_wrapper_chains=follow_wrapped)
   2752 
   2753     @property

>/usr/local/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py in _signature_from_callable(obj, follow_wrapper_chains, skip_bound_arg, sigcls)
   2336         raise ValueError(msg)
   2337 
-> 2338     raise ValueError('callable {!r} is not supported by signature'.format(obj))
   2339 
   2340 

>ValueError: callable <ufunc 'log'> is not supported by signature

The same behaviour I see also for `np.abs`.
Running freshly installed:
>hyperspy                  1.3                       (pip)
hyperspy-gui-ipywidgets   1.0.0                     (pip)

under MacOSX in Python 3.6.1
Any ideas what's wrong?

### PR DESCRIPTION
**Pull request by ericpre**
- [x] Fix #1712,
- [x] raise a warning when using map of ufuncs,
- [x] ready for review.

### CODE REVIEW

**Comment by ericpre** [comment]
I am not sure if it would be sensible to raise an error when calling map with a numpy ufunc because it would be more simple to apply the ufunc to the signal directly.

**Review by dnjohnstone — APPROVED** [review]
I think a warning is appropriate if it is only silly and doesn't break things (as this PR will cause to be true for numpy ufuncs). Whilst it might be pythonically simpler to use numpy directly here it might not be simpler to hyperspy users with limited python background. Approved despite failing tests on the basis that the Travis failure is in python 3.5 which will no longer be supported and the AppVeyor failure is not to do with this PR.

File: hyperspy/signal.py [hunk 1]
```diff
@@ -3905,12 +3905,20 @@ def map(self, function,
         # If the function has an axis argument and the signal dimension is 1,
         # we suppose that it can operate on the full array and we don't
         # iterate over the coordinates.
+        fargs = []
```
Comment (francisco-dlp) [inline hyperspy/signal.py:3908]: I would also print the exception in the log. Also, it would be good to add a test. Finally, shouldn't this go to Rnp?

Comment (ericpre) [reply hyperspy/signal.py:3908]: Should be all done now!

**Pull request merged**